pkg/smokescreen/config.go

```diff
@@ -79,6 +79,10 @@ type Config struct {
 	// ranges by default (exempting loopback and unicast ranges)
 	// This setting can be used to configure Smokescreen with a blocklist, rather than an allowlist
 	UnsafeAllowPrivateRanges bool
+
+	// Custom handler for users to allow running code per requests, users can pass in custom methods to verify requests based
+	// on headers, code for metrics etc.
+	CustomRequestHandler func(*http.Request) error
```

Comment:
Could you include what happens when an error is returned in this comment?

Reply:
for sure, done!